### PR TITLE
Add more detail to the guide for contributors

### LINKED ISSUE #213: Add PR Best Practices Section to `CONTRIBUTING.md`
I think [aspiring contributors](https://github.com/pymc-labs/pymc-marketing/issues/212) would find this very helpful.  

@drbenvincent  do you want to take a stab at this? We could base it on the PR guides for `bambi` and/or `pymc`:

https://github.com/bambinos/bambi/blob/main/docs/CONTRIBUTING.md
https://www.pymc.io/projects/docs/en/latest/contributing/pr_checklist.html
https://www.pymc.io/projects/docs/en/latest/contributing/pr_tutorial.html

### PR DESCRIPTION
**Pull request by drbenvincent**
This PR will close #213.

~~Feedback needed, in particular see parts where I've left as "????"~~

### CODE REVIEW

**Comment by drbenvincent** [comment]
I believe this is now at an acceptable standard to merge

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,54 +1,151 @@
...
-6. To check code style:
+## Local development steps
+
+1. If you have not already done so, fork the [project repository](https://github.com/pymc-labs/pymc-marketing) by clicking on the 'Fork' button near the top right of the main repository page. This creates a copy of the code under your GitHub user account.
+
+1. Clone your fork of the `pymc-marketing` repo from your GitHub account to your local disk, and add the base repository as a remote:
+
+    ```bash
+    git clone git@github.com:<your GitHub handle>/pymc-marketing.git
+    cd bambi
```
Comment (larryshamalama) [inline CONTRIBUTING.md:37]:
```suggestion
    cd pymc-marketing
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,54 +1,151 @@
...
+
+- To indicate a work in progress please mark the PR as `draft`. Drafts may be useful to (1) indicate you are working on something to avoid duplicated work, (2) request broad review of functionality or API, or (3) seek collaborators.
+
+- All other tests pass when everything is rebuilt from scratch.
+
+- When adding additional functionality, either edit an existing example, or create a new example (typically in the form of a Jupyter Notebook) in the `pymc-marketing/docs/source/mmm` or `pymc-marketing/docs/source/clv` folders. Have a look at other examples for reference. Examples should demonstrate why the new functionality is useful in practice.
+
+- Documentation and high-coverage tests are necessary for enhancements to be accepted.
+
+- Documentation follows Numpy style guide
```
Comment (larryshamalama) [inline CONTRIBUTING.md:122]:
```suggestion
- Documentation follows NumPy style guide
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (drbenvincent) [reply CONTRIBUTING.md:37]: bambi -> pymc-marketing

**Comment by drbenvincent** [comment]
Thanks @larryshamalama. Have implemented those changes

**Comment by ColtAllen** [comment]
Is there interest in adding this to the documentation?

**Comment by drbenvincent** [comment]
My preference would be no, because dev types will go to the repo and it will cut down on duplication. But certainly persuadable if people thought it was important

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,54 +1,151 @@
...
+    conda activate pymc_marketing_env
+    ```
+
+    Install the package (in editable mode) and its development dependencies:
+
+    ```bash
+    make init
+    ```
+
+    Set [pre-commit hooks](https://pre-commit.com/):
```
Comment (juanitorduz) [inline CONTRIBUTING.md:67]: they need to install `pre-commit` inside the env first. Shall we suggest `make check_lint` first? or just install `pre-commit`?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,54 +1,151 @@
-# Contributing
+# Guidelines for Contributing
 
-This document contains information about setting up a development environment for contributing to the project.
+PyMC-Marketing welcomes contributions from interested individuals or groups. These guidelines are provided to give potential contributors information to make their contribution compliant with the conventions of the PyMC-Marketing project, and maximize the probability of such contributions to be merged as quickly and efficiently as possible.
```
Comment (juanitorduz) [inline CONTRIBUTING.md:3]: Maybe something along the lines:
*You do not need to be an expert to contribute (see below), just be interested and willing to learn and share knowledge.* 
To make the guidelines more welcoming.

Comment (drbenvincent) [reply CONTRIBUTING.md]: I've added something along these lines

File: CONTRIBUTING.md (hunk 3 above)
Comment (drbenvincent) [reply CONTRIBUTING.md]: Good point. Have added something about this

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**